hal/inc/i2c_hal.h

```diff
@@ -39,8 +40,12 @@ typedef enum
 /* Exported types ------------------------------------------------------------*/
 typedef enum HAL_I2C_Interface {
     HAL_I2C_INTERFACE1 = 0,
+#if PLATFORM_ID == PLATFORM_XENON || PLATFORM_ID == PLATFORM_ARGON || PLATFORM_ID == PLATFORM_BORON
```

Comment:
This is a rebase artifact. There should no longer be a conditional here.

Reply:
Do you mean we've define all I2C instance in `hal_platform.h`, so `HAL_I2C_Init()` won't get nonexistent parameter such as `HAL_I2C_INTERFACE3`?

Reply:
It shouldn't, because we are correctly defining capabilities in `spark_wiring_platform.h` based on `HAL_PLATFORM_I2CX` defines in `hal_platform.h`

Reply:
That make sense.